web/src/typeahead_helper.ts

```diff
@@ -9,27 +9,36 @@ import * as buddy_data from "./buddy_data";
 import * as compose_state from "./compose_state";
 import {page_params} from "./page_params";
 import * as people from "./people";
+import type {PseudoMentionUser, User} from "./people";
 import * as pm_conversations from "./pm_conversations";
 import * as recent_senders from "./recent_senders";
 import * as stream_data from "./stream_data";
 import * as stream_list_sort from "./stream_list_sort";
+import type {StreamSubscription} from "./sub_store";
 import * as user_groups from "./user_groups";
+import type {UserGroup} from "./user_groups";
 import * as user_status from "./user_status";
+import type {UserStatusEmojiInfo} from "./user_status";
 import * as util from "./util";
+
+export type UserOrMention =
+    | (PseudoMentionUser & {is_broadcast: true})
+    | (User & {is_broadcast: false});
```

Comment:
`is_broadcast` was already part of the types, but I had to redefine it here to show typescript that this is a way to differentiate the union

Reply:
That’s not a thing. `PseudoMentionUser & {is_broadcast: true}` is exactly equivalent to `PseudoMentionUser`, and I don’t see any different behavior upon simplifying these lines.

Reply:
Ah okay maybe it was slow to update the warnings or something when I was playing around with this. Thanks!